webapp/src/components/AssessmentMetadataBar.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState } from 'react';
+
+import {
+  Avatar,
+  Divider,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Tooltip,
+  Typography,
+  Switch,
+} from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
+import TimerIcon from '@mui/icons-material/Timer';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import LightbulbIcon from '@mui/icons-material/Lightbulb';
+
+import { OpenedAssessment } from '../types/api.d';
+import { formatDateTime } from '../helpers/dateTime';
+
+interface AssessmentMetadataBarProps {
+  assessment: OpenedAssessment;
+  endTime: Date;
+  secondsRemaining: number | null;
+  submissionDisabled: boolean;
+}
+
+const formatTimeRemaining = (totalSeconds: number) => {
```

Comment:
You did a great job with functions and logic for retrieving, changing timestamps, and representing time on UI.